File: frontend/src/types/capitalGains.ts
export interface GainBucket {
  key: string
  label: string
  gross_gain: number
  setoff_applied: number
  exemption_applied: number
  taxable: number
  rate: number | null
  est_tax: number | null
}

export interface RealizedLot {
  symbol: string
  name: string | null
  asset_category: string
  buy_date: string
  sell_date: string
  qty: number
  buy_value: number
  sell_value: number
  expenses: number
  gain: number
  holding_days: number
  tax_bucket: string
  flags: string[]
}

export interface AttentionItem {
  symbol: string
  name: string | null
  asset_category: string
  sell_date: string
  qty: number
  sell_value: number
  flags: string[]
  reason: string
}

export interface IntradayInfo {
  trades: number
  pnl: number
}

export interface CGTotals {
  gross_gain: number
  est_tax: number
}

export interface CapitalGainsResponse {
  fy: string
  buckets: GainBucket[]
  lots: RealizedLot[]
  intraday: IntradayInfo
  attention: AttentionItem[]
  totals: CGTotals
}

export interface AvailableFYsResponse {
  fys: string[]
}

File: app/schemas/capital_gains.py
from typing import Optional
from pydantic import BaseModel


class GainBucket(BaseModel):
    key: str
    label: str
    gross_gain: float
    setoff_applied: float
    exemption_applied: float
    taxable: float
    rate: Optional[float] = None
    est_tax: Optional[float] = None


class RealizedLotOut(BaseModel):
    symbol: str
    name: Optional[str] = None
    asset_category: str
    buy_date: str
    sell_date: str
    qty: float
    buy_value: float
    sell_value: float
    expenses: float
    gain: float
    holding_days: int
    tax_bucket: str
    flags: list[str]


class AttentionItem(BaseModel):
    symbol: str
    name: Optional[str] = None
    asset_category: str
    sell_date: str
    qty: float
    sell_value: float
    flags: list[str]
    reason: str


class IntradayInfo(BaseModel):
    trades: int
    pnl: float


class CGTotals(BaseModel):
    gross_gain: float
    est_tax: float


class CapitalGainsResponse(BaseModel):
    fy: str
    buckets: list[GainBucket]
    lots: list[RealizedLotOut]
    intraday: IntradayInfo
    attention: list[AttentionItem]
    totals: CGTotals


class AvailableFYsResponse(BaseModel):
    fys: list[str]
